sphinx/transforms/post_transforms/__init__.py

```diff
@@ -171,14 +172,27 @@ def warn_missing_reference(self, refdoc: str, typ: str, target: str,
         warn = node.get('refwarn')
         if self.config.nitpicky:
             warn = True
+            dtype = '%s:%s' % (domain.name, typ) if domain else typ
             if self.config.nitpick_ignore:
-                dtype = '%s:%s' % (domain.name, typ) if domain else typ
                 if (dtype, target) in self.config.nitpick_ignore:
                     warn = False
                 # for "std" types also try without domain name
                 if (not domain or domain.name == 'std') and \
                    (typ, target) in self.config.nitpick_ignore:
                     warn = False
+            if self.config.nitpick_ignore_regex:
+                def matches_ignore(ent_typ: str, ent_tgt: str) -> bool:
```

Comment:
What the "ent" means? entry? I think names should not be abbreviated. "ign" is also.

Reply:
Sure, I changed it to be the longer versions.

About the tests, I am currently using the `filter_warnings` function to check the results. Do you think I should also copy it to `tests/test_config.py`?

Also by "I think no reason to test nitpicky_warning for each domain." do you mean that testing with both `cpp` and `c` is too much or that 2 domains is enough?

Reply:
I'm not sure it is good to reuse those test roots unless the tests are in the files. I fear the nitpick tests will falsely pass later if some refactoring is done in the domains.
How about making a new test root with a single ``index.rst`` file with specific tests instead? Then you won't need the ``filter_warnings`` function (unless you make multiple files as well).

Reply:
The best way is to add a new testroot. It would be nice if all test cases are separated. But it's okay to share the example if it's tough. So the better way is to reuse the testroot and copy the function to `test_config.py`.

>Also by "I think no reason to test nitpicky_warning for each domain." do you mean that testing with both cpp and c is too much or that 2 domains is enough?

Yes. nitpick_ignore_regex is not related to the specific domain. So no reason to test both C and C++. I think no advantage. Is there any purpose to test with multiple domains?

Reply:
I've added a separate test root and added the updated tests to `test_config.py`.